piff/input.py

```diff
@@ -979,8 +984,6 @@ def safe_to_image(wcs, ra, dec):
                     raise KeyError("Key %s not found in FITS header"%gain)
                 gain = float(header[gain])
             extra_props['gain'] = np.array([gain]*len(cat), dtype=float)
-        else:
-            extra_props['gain'] = [None] * len(cat)
```

Comment:
I don't understand this block, not because I think it is wrong, but because I don't know this code base. I am pointing this out to say that I am not qualified to review this bit I think.

Reply:
We used to always put gain in extra_props, even if it wasn't set by any of the various input options.  Now that we are writing all the properties to the HSM catalog, these show up as NaN, which is ugly and potentially confusing to users.  Same with satur.
So I changed it to not include them in the extra_props dict if they weren't defined, which just slightly changed how I check for that in the code.  But that means they don't end up getting written to the HSM catalog.

Reply:
Thanks for the explanation. Code looks good for that!